injector/stage1/src/main.rs

```diff
@@ -0,0 +1,81 @@
+use flate2::read::GzDecoder;
+use glob::glob;
+use sha2::{Digest, Sha256};
+use std::env;
+use std::fs::File;
+use std::io::Read;
+use std::io::Write;
+use std::path::PathBuf;
+use tar::Archive;
+
+// Inspired by https://medium.com/@nlauchande/rust-coding-up-a-simple-concatenate-files-tool-and-first-impressions-a8cbe680e887
+
+// read the binary contents of a file
+fn get_file(path: &PathBuf) -> std::io::Result<Vec<u8>> {
+    // open the file
+    let mut f = File::open(path)?;
+    // create an empty buffer
+    let mut buffer = Vec::new();
+
+    // read the whole file
+    match f.read_to_end(&mut buffer) {
+        Ok(_) => Ok(buffer),
+        Err(e) => Err(e),
+    }
+}
+
+// merge all given files into one buffer
+fn collect_binary_data(paths: &Vec<PathBuf>) -> std::io::Result<Vec<u8>> {
+    // create an empty buffer
+    let mut buffer = Vec::new();
+
+    // add contents of all files in paths to buffer
+    for path in paths {
+        println!("Processing {}", path.display());
+        let new_content = get_file(&path);
+        buffer
+            .write(&new_content.unwrap())
+            .expect("Could not add the file contents to the merged file buffer");
+    }
+
+    Ok(buffer)
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    // get the list of file matches to merge
+    let file_partials: Result<Vec<_>, _> = glob("zarf-payload-*")
+        .expect("Failed to read glob pattern")
+        .collect();
+
+    let mut file_partials = file_partials.unwrap();
+
+    // ensure a default sort-order
+    file_partials.sort();
+
+    // get a buffer of the final merged file contents
+    let contents = collect_binary_data(&file_partials).unwrap();
+
+    // @todo: finish testing/use sha256sum
```

Comment:
is this still todo?

Reply:
yes this has an issue I believe with the lack of a hex conversion so we aren't using it to validate in the current iteration.

Reply:
as discussed, #380 was open to cover this in a future PR